src/libcore/ptr/const_ptr.rs

```diff
@@ -295,6 +295,62 @@ impl<T: ?Sized> *const T {
         intrinsics::ptr_offset_from(self, origin)
     }
 
+    /// Returns whether two pointers are guaranteed equal. Returns `true` whenever the
+    /// pointers are known to be equal. If the pointers are equal, but the platform
+    /// isn't sure the pointers are equal, this function returns `false`.
+    ///
+    /// This function is the mirror of [`guaranteed_ne`], but not its inverse. There are pointer
+    /// comparisons for which both functions return false.
+    ///
+    /// [`guaranteed_ne`]: #method.guaranteed_ne
+    ///
+    /// # Safety
+    ///
+    /// This function may not be used to change the behaviour of any code. It may
+    /// only be used for short circuiting operations out of performance reasons.
+    /// It is unsound for this function to be used to allow safe code to observe
+    /// the return value of this function. You must be able to assume that the
+    /// function always returns `false` and your logic must still be sound.
+    /// ```
+    #[unstable(feature = "const_raw_ptr_comparison", issue = "53020")]
+    #[rustc_const_stable(feature = "const_raw_ptr_comparison", since = "1.46")]
```

Comment:
Wait, this is `rustc_const_stable`? That doesn't seem right.

Reply:
made it unstable again so we don't accidentally stabilize